test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDropDownListExtensionsTest.cs

```diff
@@ -0,0 +1,259 @@
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData["Property1"] = BasicSelectList;
+            helper.ViewData.Model = new TestModel { Property2 = "5" };
+
+            // Act
+            var dropDownListForResult = helper.DropDownListFor(m => m.Property2, new List<SelectListItem>
+            {
+                new SelectListItem { Text = "Four", Value = "4" },
+                new SelectListItem { Text = "Five", Value = "5" },
+            });
+
+            // Assert
+            Assert.Equal(expectedHtml, HtmlContentUtilities.HtmlContentToString(dropDownListForResult));
+        }
+
+        [Fact]
+        public void DropDownListFor_UsesSpecifiedSelectListAndHtmlAttributes()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property3_2_]]\" Key=\"HtmlEncode[[Value]]\" name=\"HtmlEncode[[Property3[2]]]\">" +
+                "<option selected=\"HtmlEncode[[selected]]\" value=\"HtmlEncode[[4]]\">HtmlEncode[[Four]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[5]]\">HtmlEncode[[Five]]</option>" + Environment.NewLine +
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property3 = new List<string> { "0", "2", "4" } };
+
+            // Act
+            var dropDownListForResult = helper.DropDownListFor(
+                m => m.Property3[2],
+                new List<SelectListItem>
+                {
+                    new SelectListItem { Text = "Four", Value = "4" },
+                    new SelectListItem { Text = "Five", Value = "5" },
+                },
+                new { Key = "Value", name = "CustomName" });
+
+            // Assert
+            Assert.Equal(expectedHtml, HtmlContentUtilities.HtmlContentToString(dropDownListForResult));
+        }
+
+        [Fact]
+        public void DropDownListFor_UsesSpecifiedSelectListAndOptionLabel()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property2]]\" name=\"HtmlEncode[[Property2]]\">" +
+                "<option value=\"\">HtmlEncode[[--select--]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[4]]\">HtmlEncode[[Four]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[5]]\">HtmlEncode[[Five]]</option>" + Environment.NewLine +
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData["Property1"] = BasicSelectList;
```

Comment:
Leave out; adding an `IEnumerable<SelectListItem>` for `Property1` has no impact on the HTML generated for `Property2`.